app/scripts/controllers/permissions/specifications.js

```diff
@@ -33,58 +32,18 @@ export const PermissionNames = Object.freeze({
  * PermissionController.
  */
 export const CaveatFactories = Object.freeze({
-  [CaveatTypes.restrictReturnedAccounts]: (accounts) => {
-    return { type: CaveatTypes.restrictReturnedAccounts, value: accounts };
-  },
-
-  [CaveatTypes.restrictNetworkSwitching]: (chainIds) => {
-    return { type: CaveatTypes.restrictNetworkSwitching, value: chainIds };
-  },
+  [Caip25CaveatType]: createCaip25Caveat,
 });
 
 /**
  * Gets the specifications for all caveats that will be recognized by the
  * PermissionController.
- *
- * @param {{
- *   getInternalAccounts: () => Record<string, import('@metamask/keyring-api').InternalAccount>,
- * }} options - Options bag.
  */
-export const getCaveatSpecifications = ({
-  getInternalAccounts,
-  findNetworkClientIdByChainId,
-}) => {
+export const getCaveatSpecifications = () => {
   return {
-    [CaveatTypes.restrictReturnedAccounts]: {
-      type: CaveatTypes.restrictReturnedAccounts,
-
-      decorator: (method, caveat) => {
-        return async (args) => {
-          const result = await method(args);
-          return result.filter((account) => caveat.value.includes(account));
-        };
-      },
-
-      validator: (caveat, _origin, _target) =>
-        validateCaveatAccounts(caveat.value, getInternalAccounts),
-
-      merger: (leftValue, rightValue) => {
-        const newValue = Array.from(new Set([...leftValue, ...rightValue]));
-        const diff = newValue.filter((value) => !leftValue.includes(value));
-        return [newValue, diff];
-      },
+    [Caip25CaveatType]: {
+      type: Caip25CaveatType,
```

Comment:
Have not considered it yet since we are dancing around calling `grantPermissionsIncremental` with this actual permission directly. I think we revisit adding this at the same time that we are adding the factory for the permission/caveat and making the required UI changes for it as well. Does that seem fair?